macros/audit_event_volume.sql

```diff
@@ -168,26 +168,17 @@ avarage as (
     source,
     MAX(avg_percentage) as avg_percentage,
     MAX(std_percentage) as std_percentage,
-    ARRAY_AGG(day order by day ASC) as days,
-    ARRAY_AGG(event_count order by day ASC) as event_counts,
-    ARRAY_AGG(total_source_count order by day ASC) as total_events_on_source, 
-    ARRAY_AGG(percentage order by day ASC) as percentages,
-    ARRAY_AGG(signal order by day ASC) as signals,
+    {{ avo_audit.array_agg_ordered('day', 'day', 'asc', 'days') }},
+    {{ avo_audit.array_agg_ordered('event_count', 'day', 'asc', 'event_counts') }},
+    {{ avo_audit.array_agg_ordered('total_source_count', 'day', 'asc', 'total_events_on_source') }},
+    {{ avo_audit.array_agg_ordered('percentage', 'day', 'asc', 'percentages') }}, 
+    {{ avo_audit.array_agg_ordered('signal', 'day', 'asc', 'signals') }}  
   from calculate_signal
   group by
     event_name,
     source
 )
-
-
--- Only return event_name/source combinations that have spiked or dropped for at least 1 day.
--- Disregard all combinations that were iniside the norm for the whole time period to reduce noise.
-select 
-  * 
-from aggregate_by_day_asc
-where
-    (select signal FROM UNNEST(signals) AS signal where signal = 1 GROUP BY signal) = 1
-    OR (select signal from UNNEST(signals) AS signal where signal = -1 GROUP BY signal) = -1
+{{avo_audit.find_signals('aggregate_by_day_asc')}}
```

Comment:
Nice! So here we've modified the sql to make sure it can run in both BigQuery and Snowflake? 💥

Reply:
yep! UNNEST was not possible in snowflake so we use LATERAL FLATTEN instead 👌